src/components/questionnaire/questionnaires-fill-mode.tsx

```diff
@@ -0,0 +1,77 @@
+import {
+  Card,
+  CardContent,
+  Typography,
+  Box,
+  FormControlLabel,
+  Checkbox,
+  Divider,
+  Radio
+} from "@mui/material";
+import type { AnswerOption, Question, Questionnaire } from "src/generated/homeLambdasClient";
+
+interface Props {
+  questionnaire: Questionnaire;
+  userResponses: Record<string, string[]>;
+  handleCheckboxChange: (questionText: string, answerLabel: string, isSelected: boolean) => void;
+  handleRadioChange: (questionText: string, answerLabel: string) => void;
+}
+
+const QuestionnaireFillMode = ({ questionnaire, userResponses, handleCheckboxChange, handleRadioChange }: Props) => {
+  const getCorrectAnswerCount = (question: Question) => {
+    return question.answerOptions.filter(option => option.isCorrect).length;
+  };
+
+  return (
```

Comment:
I guess this means it wasn't convenient to use the preview for create, edit and fill screens. This is fine and how you have implemented it seems fine, I do wonder if we should have some discussion about how we could make the prevew more reusable and handle the specific logic separately e.g. recording user results vs editing questions in a questionnaire. but perhaps at this point it's not worth making the change. Lets try to have some discussion about this anyway

Reply:
I left one other comment about this in other files. But this might have needed more (and better) planning from my end. 
But I would like to discuss about this and how it could be better and if it should be modifyed.